README.md

```diff
@@ -162,14 +209,60 @@ task: |
 
 # personality toolboxes map to mcp servers made available to this Agent
 toolboxes:
-  - toolboxes/echo
+  - GitHubSecurityLab/seclab-taskflow-agent/toolboxes/echo
 ```
 
+In the above, the `personality` and `task` field specifies the system prompt to be used whenever this `personality` is used.
+The `toolboxes` are the tools that are available to this `personality`. The `toolboxes` should be a list of `filekey` specifying 
+files of the `filetype` `toolbox`. 
+
+Personalities can be used in two ways. First it can be used standalone with a prompt input from the command line:
+
+```
+python3 main.py -p GitHubSecurityLab/seclab-taskflow-agent/personalities/examples/echo "echo this message"
+```
+
+In this case, `personality` and `task` from `GitHubSecurityLab/seclab-taskflow-agent/personalities/examples/echo` are used as the 
+system prompt while the user argument `echo this message` is used as a user prompt. In this use case, the only tools that this 
+personality has access to is the `toolboxes` specified in the file.
+
+Personalities can also be used in a `taskflow` to perform tasks. This is done by adding the `personality` to the `agents` field in a `taskflow` file:
+
+```yaml
+taskflow:
+  - task:
+      ...
+      agents:
+        - GitHubSecurityLab/seclab-taskflow-agent/personalities/assistant
+      user_prompt: |
+        Fetch all the open pull requests from `github/codeql` github repository. 
+        You do not need to provide a summary of the results.
+      toolboxes:
+        - GitHubSecurityLab/seclab-taskflow-agent/toolboxes/github_official
+```
+
+In this case, the `personality` specified in `agents` provides the system prompt and the user prompt is specified in `user_prompt` field of the task. A big difference in this case is that the `toolboxes` specified in the `task` will overwrite the `toolboxes` that the `personality` has accessed to. So in the above example, the `GitHubSecurityLab/seclab-taskflow-agent/personalities/assistant` will have access to the `GitHubSecurityLab/seclab-taskflow-agent/toolboxes/github_official` toolbox instead of its own toolbox. It is important to note that the `personalities` toolboxes get *overwritten* in this case, so whenever a `toolboxes` field is provided in a `task`, it'll use the provided toolboxes and `personality` loses access to its own toolboxes. e.g.
```

Comment:
```suggestion
In this case, the `personality` specified in `agents` provides the system prompt and the user prompt is specified in `user_prompt` field of the task. A big difference in this case is that the `toolboxes` specified in the `task` will overwrite the `toolboxes` that the `personality` has access to. So in the above example, the `GitHubSecurityLab/seclab-taskflow-agent/personalities/assistant` will have access to the `GitHubSecurityLab/seclab-taskflow-agent/toolboxes/github_official` toolbox instead of its own toolbox. It is important to note that the `personalities` toolboxes get *overwritten* in this case, so whenever a `toolboxes` field is provided in a `task`, it'll use the provided toolboxes and `personality` loses access to its own toolboxes. e.g.
```